src/mesh/mesh_function.C

```diff
@@ -71,7 +71,18 @@ MeshFunction::MeshFunction (const EquationSystems & eqn_systems,
 {
 }
 
-
+MeshFunction::MeshFunction (const MeshFunction & mf):
+  FunctionBase<Number> (mf._master),
+  ParallelObject       (mf._eqn_systems),
+  _eqn_systems         (mf._eqn_systems),
+  _vector              (mf._vector),
+  _dof_map             (mf._dof_map),
+  _system_vars         (mf._system_vars),
+  _out_of_mesh_mode    (false)
```

Comment:
`_out_of_mesh_mode` needs to be copied from the incoming `MeshFunction`, not hard-coded to `false`.

Reply:
Agreed, but take a look at our existing code too and you can see what went wrong - he's not adding a new bug here, he's retaining an existing bug.

Reply:
> he's retaining an existing bug.

I guess he copy-pasted from one of the existing constructors...? There I'd argue it's not a bug to hard-code `_out_of_mesh_mode` to false.

Reply:
Maybe, but I'm not just talking about the existing constructors, rather about the existing ```clone()```.  It doesn't pass ```_out_of_mesh_mode``` to the constructor used there (because there's no constructor that takes it) but it also doesn't use a setter to change ```_out_of_mesh_mode``` afterwards.

Reply:
Ooh, and this appears to be in the "oldest libMesh bugs" Hall Of Fame.  I added FunctionBase::clone() and implementations in 9a3d8b10308cfb247c55d8c7abcb05f13488bda3 in Feb 2012, and at that time ```_out_of_mesh_mode``` already existed yet somehow I just ignored it.

Reply:
Sorry, this is terrible. Was a bit late in the day.

Reply:
Nah, this is a perfectly understandable bug.  (where "perfectly understandable" is super-conveniently defined as "I wrote the same bug too")  Feel free to give yourself hell over the other clone() bug, though.  :wink: